contracts/mirror/src/contract.rs

```diff
@@ -93,37 +107,90 @@ pub fn execute(
     msg: ExecuteMsg,
 ) -> ContractResult<Response<NeutronMsg>> {
     match msg {
-        ExecuteMsg::Bond {
-            receiver,
-            backup,
-            r#ref,
-        } => execute_bond(deps, env, info, receiver, r#ref, backup),
+        ExecuteMsg::Bond { receiver, r#ref } => execute_bond(deps, env, info, receiver, r#ref),
         ExecuteMsg::UpdateOwnership(action) => {
             update_ownership(deps.into_empty(), &env.block, &info.sender, action)?;
             Ok(Response::new())
         }
-        ExecuteMsg::Complete { items } => execute_complete(deps, env, info, items),
-        ExecuteMsg::ChangeReturnType { id, return_type } => {
-            execute_change_return_type(deps, env, info, id, return_type)
-        }
-        ExecuteMsg::UpdateBond {
-            id,
-            receiver,
-            backup,
-            return_type,
-        } => execute_update_bond(deps, env, info, id, receiver, backup, return_type),
         ExecuteMsg::UpdateConfig { new_config } => execute_update_config(deps, info, new_config),
+        ExecuteMsg::Retry { receiver } => execute_retry(deps, env, receiver),
     }
 }
 
-pub fn execute_update_config(
+fn execute_retry(
+    deps: DepsMut<NeutronQuery>,
+    env: Env,
+    receiver: String,
+) -> ContractResult<Response<NeutronMsg>> {
+    let failed_transfers = FAILED_TRANSFERS.may_load(deps.storage, receiver.clone())?;
+    let Config {
+        source_port,
+        source_channel,
+        ibc_timeout,
+        retry_limit,
+        ..
+    } = CONFIG.load(deps.storage)?;
+    let mut ibc_transfer_submsgs: Vec<SubMsg<NeutronMsg>> = vec![];
+    let mut attrs: Vec<Attribute> = vec![attr("action", "execute_retry")];
+    if let Some(failed_transfers) = failed_transfers {
```

Comment:
as mentioned before: pull one, send it, reduce array and that is it. So please make it one at a time